src/doc/guide.md

```diff
@@ -642,21 +633,20 @@ let x = (let y = 5i); // expected identifier, found keyword `let`
 The compiler is telling us here that it was expecting to see the beginning of
 an expression, and a `let` can only begin a statement, not an expression.
 
-Note that assigning to an already-bound variable (e.g. `y = 5i`) is still an
-expression, although its value is not particularly useful. Unlike C, where an
-assignment evaluates to the assigned value (e.g. `5i` in the previous example),
-in Rust the value of an assignment is the unit type `()` (which we'll cover later).
+> Note that assigning to an already-bound variable (e.g. `y = 5i`) is still an
+> expression, although its value is not particularly useful. Unlike C, where an
+> assignment evaluates to the assigned value (e.g. `5i` in the previous
+> example), in Rust the value of an assignment is the unit type `()` (which
+> we'll cover later).
 
-The second kind of statement in Rust is the **expression statement**. Its
-purpose is to turn any expression into a statement. In practical terms, Rust's
-grammar expects statements to follow other statements. This means that you use
-semicolons to separate expressions from each other. This means that Rust
-looks a lot like most other languages that require you to use semicolons
-at the end of every line, and you will see semicolons at the end of almost
-every line of Rust code you see.
+The second kind of statement in Rust is the **expression statement**. We turn
+expressions into expression statements by placing semicolons at the end of
+expressions. The purpose is to trigger any side effects of evaluating the
```

Comment:
How about this?
"When an expression statement is evaluated, its side effects are triggered and the value of the expression is discarded."